routers/user/home.go

```diff
@@ -215,93 +219,121 @@ func Issues(ctx *context.Context) {
-			repo.NumOpenIssues, repo.NumClosedIssues = int(numOpen), int(numClosed)
-		}
-
-		if repo.ID == repoID ||
-			(isShowClosed && repo.NumClosedIssues > 0) ||
-			(!isShowClosed && repo.NumOpenIssues > 0) {
-			showRepos = append(showRepos, repo)
-		}
+		userRepoIDs = append(userRepoIDs, repo.ID)
 	}
-	ctx.Data["Repos"] = showRepos
 
-	issueStats := models.GetUserIssueStats(repoID, ctxUser.ID, repoIDs, filterMode, isPullList)
-	issueStats.AllCount = int64(allCount)
-
-	page := ctx.QueryInt("page")
-	if page <= 1 {
-		page = 1
+	var issues []*models.Issue
+	switch filterMode {
+	case models.FM_YOUR_REPOSITORIES:
+		// Get all issues from repositories from this user.
+		issues, err = models.Issues(&models.IssuesOptions{
+			RepoIDs:  userRepoIDs,
+			RepoID:   repoID,
+			Page:     page,
+			IsClosed: isShowClosed,
+			IsPull:   isPullList,
+			SortType: sortType,
+		})
+
+	case models.FM_ASSIGN:
+		// Get all issues assigned to this user.
+		issues, err = models.Issues(&models.IssuesOptions{
+			RepoID:     repoID,
+			AssigneeID: ctxUser.ID,
+			Page:       page,
+			IsClosed:   isShowClosed,
+			IsPull:     isPullList,
+			SortType:   sortType,
+		})
+
+	case models.FM_CREATE:
+		// Get all issues created by this user.
+		issues, err = models.Issues(&models.IssuesOptions{
+			RepoID:   repoID,
+			PosterID: ctxUser.ID,
+			Page:     page,
+			IsClosed: isShowClosed,
+			IsPull:   isPullList,
+			SortType: sortType,
+		})
 	}
 
-	var total int
-	if !isShowClosed {
-		total = int(issueStats.OpenCount)
-	} else {
-		total = int(issueStats.ClosedCount)
-	}
-	ctx.Data["Page"] = paginater.New(total, setting.UI.IssuePagingNum, page, 5)
-
-	// Get issues.
-	issues, err := models.Issues(&models.IssuesOptions{
-		UserID:     ctxUser.ID,
-		AssigneeID: assigneeID,
-		RepoID:     repoID,
-		PosterID:   posterID,
-		RepoIDs:    repoIDs,
-		Page:       page,
-		IsClosed:   isShowClosed,
-		IsPull:     isPullList,
-		SortType:   sortType,
-	})
 	if err != nil {
 		ctx.Handle(500, "Issues", err)
 		return
 	}
 
-	// Get posters and repository.
-	for i := range issues {
-		issues[i].Repo, err = models.GetRepositoryByID(issues[i].RepoID)
+	showRepos := make([]*models.Repository, 0, len(issues))
+	showReposSet := make(map[int64]bool)
+
+	if repoID > 0 {
+		repo, err := models.GetRepositoryByID(repoID)
 		if err != nil {
-			ctx.Handle(500, "GetRepositoryByID", fmt.Errorf("[#%d]%v", issues[i].ID, err))
+			ctx.Handle(500, "GetRepositoryByID", fmt.Errorf("[#%d]%v", repoID, err))
 			return
 		}
 
-		if err = issues[i].Repo.GetOwner(); err != nil {
-			ctx.Handle(500, "GetOwner", fmt.Errorf("[#%d]%v", issues[i].ID, err))
+		if err = repo.GetOwner(); err != nil {
+			ctx.Handle(500, "GetOwner", fmt.Errorf("[#%d]%v", repoID, err))
 			return
 		}
+
+		showReposSet[repoID] = true
+		showRepos = append(showRepos, repo)
 	}
-	ctx.Data["Issues"] = issues
 
+	showIssues := make([]*models.Issue, 0, len(issues))
```

Comment:
This block of code is not right.

We must control returned issues when we call`models.Issues`, not filter after that. Otherwise pagination is broken.